### DISCUSSION TITLE
Logging settings by proxy host + modsecurity blocking log check function suggestion

### DISCUSSION THREAD

**Discussion by dalso0418** [ZoeyVid/NPMplus#503] [Q&A] [answered]
Hello. It's been a while since I said hello.
Is it possible to add an option to leave nginx log like the existing npm? It would be nice to add an option for log to leave separately even if it is not for proxy host.

In addition, it would be fantastic if there was an option to view logs blocked by modsecurity in webui and make exceptions for that rule id.

Thank you for the good project :)

**Comment by Zoey2936** [comment]
For the first, I think you mean access logs, but sorry, I will not add them back, because they would break EU law.

For the seccond, because they are logged to the docker logs, NPMplus would need to have access to the docker socket to read them, which is too unsecure from my view.
NPMplus also doesn't log all IDs which causes the block to the console, since without debug logging this is not possible (which would be too much of logging). But I can make the REQUEST-900-EXCLUSION-RULES-BEFORE-CRS.conf.example and RESPONSE-999-EXCLUSION-RULES-AFTER-CRS.conf.example editable

**Comment by Zoey2936** [comment]
added in new release

**Comment by fa-ve** [comment]
There is a way to do IP masking in NGNIX,: https://www.nginx.com/blog/data-masking-user-privacy-nginscript/ it would be useful for troubleshooting to have some logs. As I understand you're from Germany and the privacy rules are even stronger than in the rest of Europe

**Comment by Zoey2936** [comment]
That would be an option... 
But I just want to mention that you have the custom folder in /opt/npm/nginx, where you can enable the access_log for yourself. 
But if I add an option like mentioned by you, I would add only one log file and not for each nginx host one, and also I wouldn't add any log rotation.

**Comment by fa-ve** [comment]
Can you explain how? where you can enable the access_log for yourself 
As I'm still leaning :)

As I want to use when needed: https://github.com/xavier-hernandez/goaccess-for-nginxproxymanager but then I need log files.

**Comment by Zoey2936** [comment, marked as answer]
in the `/opt/npmplus/nginx/custom_*.conf` files you can configure the logs how you want: https://nginx.org/en/docs/http/ngx_http_log_module.html

**Reply by dalso0418** [reply]
Hello, I tried to apply the log as you said, but it doesn't work normally, so I'm asking you an additional question.

vi /opt/npmplus/nginx/custom_*.conf

The content is written as below, but it doesn't work properly. Restarting npmplus converts to 0byte.
+ I tried custom/log.conf, but it's the same.

access_log /var/log/nginx/access.log;
error_log /var/log/nginx/error.log;

**Reply by Zoey2936** [reply]
you need to use a path which is mounted, for example: /data/log.txt

**Reply by dalso0418** [reply]
Oh, I'm sorry.
The mount path was done as /data/logs/access.log as shown above.

Where should I put the set value location?

**Reply by dalso0418** [reply]
Even if you modify conf in custom, restarting npmplus will all initialize and will not be logged.

**Reply by Zoey2936** [reply]
how does your conf files look like?

**Reply by dalso0418** [reply]
I've added the content to conf in the custom folder and created a new log.conf, but if you re-run them all, it will be initialized to 0byte.

![image](https://github.com/ZoeyVid/NPMplus/assets/107911055/ce659ccb-eac9-4719-a023-f79ce9829eeb)

**Reply by Zoey2936** [reply]
so the files will be deleted?

**Reply by dalso0418** [reply]
If you restart the container, the file will not be deleted, but all contents will disappear and become zero bytes.

**Reply by Zoey2936** [reply]
I've found the error, will release now

**Reply by dalso0418** [reply]
Thank you I will check it on the weekend :)

**Comment by dalso0418** [comment]
Thank you for your reply :) I'll check it out!
